scrapy/robotstxt.py

```diff
@@ -0,0 +1,113 @@
+import sys
+import logging
+from abc import ABCMeta, abstractmethod
+
+from scrapy.utils.python import to_native_str, to_unicode
+
+logger = logging.getLogger(__name__)
+
+
+class RobotParser:
+    __metaclass__ = ABCMeta
```

Comment:
This syntax only works in Python 2, six.with_metaclass should be used if you want code to be compatible with both Python 2 and 3.

Reply:
I made the change.